### PR TITLE
build(ot): convert to fluid-tsc

### PR DESCRIPTION
**Pull request by DLehenbauer**
READY.
█

### CODE REVIEW

File: experimental/dds/ot/ot/package.json [hunk 1]
```diff
@@ -11,18 +11,55 @@
...
 		"build:commonjs": "fluid-build . --task commonjs",
 		"build:compile": "fluid-build . --task compile",
-		"build:docs": "api-extractor run --local",
-		"build:esnext": "tsc-multi --config ../../../../common/build/build-common/tsc-multi.esm.json",
+		"build:docs": "fluid-build . --task api",
+		"build:esnext": "tsc --project ./tsconfig.json",
 		"build:genver": "gen-version",
-		"build:test": "tsc --project ./src/test/tsconfig.json",
-		"check:are-the-types-wrong": "attw --pack",
+		"build:test": "npm run build:test:cjs && npm run build:test:esm",
```
Comment (jason-ha) [inline experimental/dds/ot/ot/package.json:59]: nit: a little nicer to build esm first. It is more restrictive and these will run in order. So, we prefer the esm build first.

File: experimental/dds/ot/ot/src/test/delta.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import {
 	IChannelFactory,
 } from "@fluidframework/datastore-definitions";
 import Delta from "quill-delta";
-import { SharedOT } from "../../dist";
+import { SharedOT } from "../../dist/index.js";
```
Comment (jason-ha) [inline experimental/dds/ot/ot/src/test/delta.ts:13]: eh? a dist reference?

**Review by jason-ha — CHANGES_REQUESTED** [review]
Might be useful to understand how the test dist reference works well in the ESM case.

File: experimental/dds/ot/ot/src/test/delta.ts (hunk 2 above)
Comment (DLehenbauer) [reply experimental/dds/ot/ot/src/test/delta.ts]: Weird.  Here is what ../../dist/index.js looks like:
```
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedOT = void 0;
var ot_js_1 = require("./ot.js");
Object.defineProperty(exports, "SharedOT", { enumerable: true, get: function () { return ot_js_1.SharedOT; } });
//# sourceMappingURL=index.js.map
```

Comment (jason-ha) [reply experimental/dds/ot/ot/src/test/delta.ts]: That looks like ESM compatible CJS to me. The presence of package.json in dist allows it to be viable, but we shouldn't reference it without good reason and comments. Perhaps just some obsolete reason now.

**Review by jason-ha — APPROVED** [review]

File: experimental/dds/ot/ot/src/test/delta.ts (hunk 2 above)
Comment (DLehenbauer) [reply experimental/dds/ot/ot/src/test/delta.ts]: Yeah, I'm sure it was accidental...  VS Code will sometimes helpfully generate imports like this when the language service is confused/reparsing.  Fixed now.

**Pull request merged**